admin/admin.go

```diff
@@ -0,0 +1,178 @@
+package admin
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/vocdoni/blind-csp/model"
+	"go.vocdoni.io/dvote/httprouter"
+	"go.vocdoni.io/dvote/httprouter/bearerstdapi"
+	"go.vocdoni.io/dvote/log"
+)
+
+type Admin struct {
+	router             *httprouter.HTTProuter
+	api                *bearerstdapi.BearerStandardAPI
+	storage            *model.MongoStorage
+	electionController *ElectionController
+	userController     *UserController
+}
+
+func NewAdmin() (*Admin, error) {
+	admin := new(Admin)
+
+	var err error
+
+	// Storage
+	admin.storage = &model.MongoStorage{}
+	if err := admin.storage.Init(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Models & Controllers
+	electionStore := model.NewElectionStore(admin.storage)
+	admin.electionController = NewElectionController(electionStore)
+
+	userStore := model.NewUserStore(admin.storage)
+	admin.userController = NewUserController(userStore)
+
+	return admin, err
+}
+
+func (admin *Admin) ServeAPI(r *httprouter.HTTProuter, baseRoute string) error {
+	var authToken []string
+	if len(authToken) == 0 {
+		authToken = strings.Split(os.Getenv("ADMINAPI_AUTHTOKEN"), ",")
+	}
+
+	if len(baseRoute) == 0 || baseRoute[0] != '/' {
+		return fmt.Errorf("invalid base route (%s), it must start with /", baseRoute)
+	}
+
+	// Remove trailing slash
+	if len(baseRoute) > 1 {
+		baseRoute = strings.TrimSuffix(baseRoute, "/")
+	}
+	if r == nil {
+		return fmt.Errorf("router is nil")
+	}
+
+	// Initialize API
+	admin.router = r
+	var err error
+	admin.api, err = bearerstdapi.NewBearerStandardAPI(admin.router, baseRoute)
+	if err != nil {
+		return err
+	}
+
+	// Create the Bearer API attached to the router
+	admin.api.EnableVerboseAuthLog()
+
+	// Set bearer authentication
+	if len(authToken) == 0 || authToken[0] == "" {
+		authToken = []string{uuid.New().String()}
+	}
+	for _, at := range authToken {
+		admin.api.AddAuthToken(at, 100000000)
+	}
+	log.Infof("using bearer authentication token %s", authToken)
+
+	return admin.registerHandlers()
+}
+
+func (admin *Admin) registerHandlers() error {
+	if err := admin.api.RegisterMethod(
+		"/elections",
+		"GET",
+		bearerstdapi.MethodAccessTypePublic,
+		admin.electionController.List,
+	); err != nil {
+		return err
+	}
+
+	// Create census and link it to a new election
+	if err := admin.api.RegisterMethod(
+		"/elections",
+		"POST",
+		bearerstdapi.MethodAccessTypePublic,
+		admin.electionController.Create,
```

Comment:
I think Create, Delete, etc, should keep private

Reply:
Currently authentication is performed using a manual election-based/Bearer token (not supported in bearerstdapi). This is a temp. solution, that was know and has already been decided to change.
Agree that next future-proof mechanism should be implemented in "middleware" layer (bearerstdapi or similar) and marked as not public.

Reply:
Actually, reviewing a little deeply the code I think all the methods inside `election_controller` and `user_controller` could be private without problem.

Reply:
Auth will be handled in the future. This is a PoC implementation.